core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -142,6 +141,10 @@ case class FetchPartitionData(error: Errors = Errors.NONE,
                               preferredReadReplica: Option[Int],
                               isReassignmentFetch: Boolean)
 
+case class FencedPartitionMetadata(partition: Partition,
```

Comment:
nit: seems like we can flatten this into `Fenced`. Otherwise it is a case class consisting of only another case class.